### ISSUE TITLE
FTBFS with GCC 14 due to implicit declaration of function

### ISSUE THREAD

**Issue by nanake** [zapping-vbi/zvbi#43]
```
$CC --version
x86_64-w64-mingw32ucrt-gcc (GCC) 14.0.1 20240127 (Fedora MinGW 14.0.1-1.fc40)
```

```bash
/bin/sh ../libtool  --tag=CC   --mode=compile x86_64-w64-mingw32ucrt-gcc -DHAVE_CONFIG_H -I. -I..  -I..  -D_REENTRANT -D_GNU_SOURCE  --param max-inline-insns-single=50000 --param large-function-insns=150000 --param inline-unit-growth=30000 --param large-function-growth=30000 -static-libgcc -static-libstdc++ -I/opt/ffbuild/include -O2 -pipe -D_FORTIFY_SOURCE=2 -fstack-protector-strong -MT libzvbiinline_la-exp-gfx.lo -MD -MP -MF .deps/libzvbiinline_la-exp-gfx.Tpo -c -o libzvbiinline_la-exp-gfx.lo `test -f 'exp-gfx.c' || echo './'`exp-gfx.c
In file included from /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/_mingw.h:10,
                 from /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/corecrt.h:10,
                 from /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/crtdefs.h:10,
                 from /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/string.h:9,
                 from /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/sec_api/string_s.h:9,
                 from LibOb_strptime.c:143:
/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/string.h: In function 'wcscpy':
/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/string.h:268:5: error: implicit declaration of function 'wcscpy_s'; did you mean 'wcscpy'? [-Wimplicit-function-declaration]
  268 |     __mingw_bos_cond_chk(!wcscpy_s(__dst, __mingw_bos(__dst, 1) / sizeof(wchar_t), __src));
      |     ^~~~~~~~~~~~~~~~~~~~
/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/string.h: In function 'wcscat':
/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include/string.h:278:5: error: implicit declaration of function 'wcscat_s'; did you mean 'wcscat'? [-Wimplicit-function-declaration]
  278 |     __mingw_bos_cond_chk(!wcscat_s(__dst, __mingw_bos(__dst, 1) / sizeof(wchar_t), __src));
      |     ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:800: LibOb_strptime.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  x86_64-w64-mingw32ucrt-gcc -DHAVE_CONFIG_H -I. -I.. -I.. -D_REENTRANT -D_GNU_SOURCE --param max-inline-insns-single=50000 --param large-function-insns=150000 --param inline-unit-growth=30000 --param large-function-growth=30000 -static-libgcc -static-libstdc++ -I/opt/ffbuild/include -O2 -pipe -D_FORTIFY_SOURCE=2 -fstack-protector-strong -MT libzvbiinline_la-decoder.lo -MD -MP -MF .deps/libzvbiinline_la-decoder.Tpo -c decoder.c -o libzvbiinline_la-decoder.o
mv -f .deps/wss.Tpo .deps/wss.Plo
mv -f .deps/io-sim.Tpo .deps/io-sim.Plo
mv -f .deps/vbi.Tpo .deps/vbi.Plo
libtool: compile:  x86_64-w64-mingw32ucrt-gcc -DHAVE_CONFIG_H -I. -I.. -I.. -D_REENTRANT -D_GNU_SOURCE --param max-inline-insns-single=50000 --param large-function-insns=150000 --param inline-unit-growth=30000 --param large-function-growth=30000 -static-libgcc -static-libstdc++ -I/opt/ffbuild/include -O2 -pipe -D_FORTIFY_SOURCE=2 -fstack-protector-strong -MT libzvbiinline_la-exp-gfx.lo -MD -MP -MF .deps/libzvbiinline_la-exp-gfx.Tpo -c exp-gfx.c -o libzvbiinline_la-exp-gfx.o
mv -f .deps/trigger.Tpo .deps/trigger.Plo
mv -f .deps/xds_demux.Tpo .deps/xds_demux.Plo
mv -f .deps/teletext.Tpo .deps/teletext.Plo
mv -f .deps/packet.Tpo .deps/packet.Plo
mv -f .deps/inout.Tpo .deps/inout.Plo
mv -f .deps/libzvbiinline_la-decoder.Tpo .deps/libzvbiinline_la-decoder.Plo
mv -f .deps/libzvbiinline_la-exp-gfx.Tpo .deps/libzvbiinline_la-exp-gfx.Plo
mv -f .deps/pdc.Tpo .deps/pdc.Plo
make[2]: Leaving directory '/zvbi/src'
make[1]: *** [Makefile:854: all-recursive] Error 1
make[1]: Leaving directory '/zvbi/src'
make: *** [Makefile:636: all] Error 2
make: Leaving directory '/zvbi/src'
```

**Comment by ileanadumitrescu95** [comment]
I am not able to reproduce this FTBFS. However, if you can offer assistance in testing some proposed fixes, we may be able to fix this together by trying each of the following:

1) Define \_\_STDC_WANT_LIB_EXT1\_\_ in LibOb_strptime.c before the include for <sec_api/string_s.h> [1].
```
 // For later extension (ms):
 // SYSTEMTIME st;
 // GetSystemTime(&st);
 // vsnprintf_s
 
+#define __STDC_WANT_LIB_EXT1__
 #include <sec_api/string_s.h>
 #include <stdio.h>
 #include "LibOb_strptime.h"
```
2) Specify ```-DMINGW_HAS_SECURE_API=1``` on compilation [2].

If neither of the above work, it may be worth making a separate issue with mingw.

[1] https://en.cppreference.com/w/c/string/wide/wcscpy
[2] https://github.com/godotengine/godot/issues/33062

**Comment by nanake** [comment]
Are you sure you compile with mingw-gcc-14 where turning `-Werror=implicit-function-declaration` by default? https://discourse.nixos.org/t/rfc-more-c-errors-by-default-in-gcc-14/27390

Both methods don't work for me. 

Anyway, I can restore old behavior using `-Wno-error=implicit-function-declaration` or `-fpermissive` flag when compile.

**Comment by ileanadumitrescu95** [comment]
I am happy there are options to restore old behaviour, but I have not reproduced this FTBFS since mingw-w64 with gcc-14 is not available in Debian yet. For now, I have created a pull request [1] where you can test another proposed fix.

[1] [zvbi pull request #45](https://github.com/zapping-vbi/zvbi/pull/45)

**Comment by ileanadumitrescu95** [comment]
Resolved in commit [e0ff90c](https://github.com/zapping-vbi/zvbi/commit/e0ff90c338500fd468c7925db85bed4b1b2d1e72)
